src/Composer/ComposerWidget.vala

```diff
@@ -787,4 +831,41 @@ public class Mail.ComposerWidget : Gtk.Grid {
             set_css_name (Gtk.STYLE_CLASS_ENTRY);
         }
     }
+
+    public override void destroy () {
+        if (discard_draft) {
+            base.destroy ();
+            return;
+        }
+
+        web_view.get_body_html.begin ((obj, res) => {
+            var body_html = web_view.get_body_html.end (res);
+
+            if (body_html == null || get_first_line_without_html (body_html) == first_line_of_message_without_html_on_load) {
+                base.destroy ();
+
+            } else {
+                unowned Mail.Backend.Session session = Mail.Backend.Session.get_default ();
+
+                var message = build_message (body_html);
+                var sender = build_sender (message, from_combo.get_active_text ());
+                var recipients = build_recipients (message, to_val.text, cc_val.text, bcc_val.text);
+
+                session.save_draft.begin (
+                    message,
+                    sender,
+                    recipients,
+                    (obj, res) => {
+                        try {
+                            session.save_draft.end (res);
+                        } catch (Error e) {
+                            critical ("Unable to safe draft: %s", e.message);
+                            // TODO: Shall we display a dialog here?
```

Comment:
Yeah this should definitely not silently fail and in fact we should probably not quit the composer if this returns an error to avoid data loss

Reply:
I implemented a re-rendering of the ComposerWidget within a new ComposerWindow in case saving the draft fails. In addition, an error dialog is shown now.